test/article.test.js

```diff
@@ -2,16 +2,51 @@ const chai = require('chai')
 const chaiHttp = require('chai-http')
 const app = require('../server.js')
 
-chai.should()
-chai.use(chaiHttp)
+const should = chai.should()
 
-function checkResponse(res) {
+function postVote(id, vote, old) {
+  return new Promise((resolve) => {
+    chai.request(app)
+            .post('/articles/vote')
+            .send({ id, vote })
+            .end((err, res) => {
+              resolve({ res, old })
+            })
+  })
+}
+
+function getArticleData(id) {
+  return new Promise((resolve, reject) => {
+    chai.request(app)
+            .get(`/articles/${id}`)
+            .end((err, res) => {
+              if (err) reject(err)
+              resolve(res)
+            })
+  })
+}
+
+function checkReponse(res) {
   res.should.have.status(200)
   res.body.should.be.an('array')
   res.body.should.not.be.empty
   res.body.length.should.be.at.most(10)
 }
 
+function checkValidError(res) {
+  res.should.have.status(500)
```

Comment:
We should check if it's not 200 just incase we decide to change the error statuses sent out, I think they should be in the 400's